packages/elasticsearch-asset-apis/test/api-with-dataFrams-spec.ts

```diff
@@ -0,0 +1,484 @@
+
+            const api = await createElasticsearchReaderAPI({
+                config, client, logger, emitter
+            });
+
+            const results = await api.determineDateRanges();
+
+            expect(results).toBeDefined();
+            expect(results.start).toBeDefined();
+            expect(results.limit).toBeDefined();
+
+            expect(results.start?.toISOString()).toEqual('2019-04-26T15:00:23.201Z');
+            expect(results.limit?.toISOString()).toEqual('2019-04-26T15:00:23.394Z');
+        });
+
+        it('can determine slice interval', async () => {
+            const config: ESReaderOptions = {
+                ...defaultConfig
+            };
+
+            const api = await createElasticsearchReaderAPI({
+                config, client, logger, emitter
+            });
+
+            const dates = await api.determineDateRanges() as InputDateSegments;
+            const results = await api.determineSliceInterval(config.interval, dates);
+
+            expect(results).toBeDefined();
+            expect(results).toBeArrayOfSize(2);
+            expect(results).toEqual([193, 'ms']);
+        });
+
+        it('can make date slices', async () => {
+            const config: ESReaderOptions = {
+                ...defaultConfig
+            };
+
+            const api = await createElasticsearchReaderAPI({
+                config, client, logger, emitter
+            });
+
+            const slicer = await api.makeDateSlicer({
+                lifecycle: 'once',
+                slicerID: 0,
+                numOfSlicers: 1,
+                recoveryData: [],
+            });
+
+            expect(slicer).toBeDefined();
+
+            const slice = await slicer() as SlicerDateResults;
+
+            expect(slice).toBeDefined();
+            expect(slice.start).toEqual('2019-04-26T15:00:23.201Z');
+            expect(slice.end).toEqual('2019-04-26T15:00:23.394Z');
+            expect(slice.limit).toEqual('2019-04-26T15:00:23.394Z');
+            expect(slice.holes).toEqual([]);
+            expect(slice.count).toEqual(1000);
+
+            const nullSlice = await slicer();
+            expect(nullSlice).toBeNull();
+        });
+
+        it('can count a slice', async () => {
+            const config: ESReaderOptions = {
+                ...defaultConfig
+            };
+
+            const api = await createElasticsearchReaderAPI({
+                config, client, logger, emitter
+            });
+            // query is set to * above
+            const count = await api.count();
+
+            expect(count).toEqual(1000);
+        });
+
+        it('can fetch records', async () => {
+            const config: ESReaderOptions = {
+                ...defaultConfig
+            };
+
+            const api = await createElasticsearchReaderAPI({
+                config, client, logger, emitter
+            });
+
+            const results = await api.fetch() as DataFrame;
+
+            expect(results).toBeDefined();
```

Comment:
This should ensure the results is instanceof DataFrame.

```suggestion
            expect(results).toBeInstanceOf(DataFrame);
```